tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/AssemblyResolverCtorIntegration.cs

```diff
@@ -0,0 +1,85 @@
+﻿// <copyright file="AssemblyResolverCtorIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing;
+
+/// <summary>
+/// System.Void Microsoft.VisualStudio.TestPlatform.Common.Utilities.AssemblyResolver::.ctor(System.Collections.Generic.IEnumerable`1[System.String]) calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "Microsoft.VisualStudio.TestPlatform.Common",
+    TypeName = "Microsoft.VisualStudio.TestPlatform.Common.Utilities.AssemblyResolver",
+    MethodName = ".ctor",
+    ReturnTypeName = ClrNames.Void,
+    ParameterTypeNames = ["System.Collections.Generic.IEnumerable`1[System.String]"],
+    MinimumVersion = "15.0.0",
+    MaximumVersion = "15.*.*",
+    IntegrationName = IntegrationName)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public class AssemblyResolverCtorIntegration
+{
+    private const string IntegrationName = "TestPlatformAssemblyResolver";
+
+    private static TaskCompletionSource<bool>? _callHasBeenCompletedTaskCompletionSource;
+
+    /// <summary>
+    /// OnMethodBegin callback
+    /// </summary>
+    /// <typeparam name="TTarget">Type of the target</typeparam>
+    /// <typeparam name="TArg1">Type of the argument 1 (System.Collections.Generic.IEnumerable`1[System.String])</typeparam>
+    /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+    /// <param name="directories">Instance of System.Collections.Generic.IEnumerable`1[System.String]</param>
+    /// <returns>Calltarget state value</returns>
+    internal static CallTargetState OnMethodBegin<TTarget, TArg1>(TTarget instance, ref TArg1 directories)
+    {
+        Common.Log.Debug("Microsoft.VisualStudio.TestPlatform.Common.Utilities.AssemblyResolver.ctor started.");
+        _callHasBeenCompletedTaskCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Is there any risk of `new AssemblyResolver()` being called more than once? 🤔

Reply:
It's called twice but for different scenarios, one as of them for the IDE an scenario that is not covered by CI Visibility. So we are safe.